lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -70,6 +71,37 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  def validate_previous_transaction_type?(tx) do
+    case Transaction.network_type?(tx.type) && tx.type not in [:code_proposal] do
+      false ->
+        #  not a network tx , no need to validate with
+        #  last tx
+        true
+
+      true ->
+        # when network tx , check with previous transaction
+        do_validate_previous_transaction_type?(tx)
+    end
+  end
+
+  def do_validate_previous_transaction_type?(tx) do
+    case TransactionChain.get_last_transaction(tx.address, [:address, :type]) do
```

Comment:
I think you can just get the previous address from: `Transaction.previous_address(tx)` which is calculating the hash from the previous public key, then apply a verification if the type is not the same

Reply:
You just need to be careful because some network transaction chain can have multiple types:
- oracles (oracle, oracle_summary)
- rewards (mint_rewards, node_rewards)